### PR TITLE
Fixes #8508 spuriously failing

### LINKED ISSUE #8508: [map] [map-store] MapLoaderTest.testSenderAndBackupTerminates_AfterInitialLoad
```
org.junit.runners.model.TestTimedOutException: test timed out after 300000 milliseconds
    at org.apache.log4j.Category.callAppenders(Category.java:201)
    at org.apache.log4j.Category.forcedLog(Category.java:388)
    at org.apache.log4j.Category.log(Category.java:823)
    at com.hazelcast.logging.Log4jFactory$Log4jLogger.log(Log4jFactory.java:50)
    at com.hazelcast.logging.LoggingServiceImpl$DefaultLogger.log(LoggingServiceImpl.java:167)
    at com.hazelcast.logging.LoggingServiceImpl$DefaultLogger.log(LoggingServiceImpl.java:155)
    at com.hazelcast.logging.AbstractLogger.fine(AbstractLogger.java:45)
    at com.hazelcast.test.mocknetwork.MockJoiner.getJoinAddress(MockJoiner.java:84)
    at com.hazelcast.test.mocknetwork.MockJoiner.doJoin(MockJoiner.java:51)
    at com.hazelcast.internal.cluster.impl.AbstractJoiner.join(AbstractJoiner.java:135)
    at com.hazelcast.instance.Node.join(Node.java:650)
    at com.hazelcast.instance.Node.start(Node.java:362)
    at com.hazelcast.instance.HazelcastInstanceImpl.<init>(HazelcastInstanceImpl.java:132)
    at com.hazelcast.instance.HazelcastInstanceFactory.constructHazelcastInstance(HazelcastInstanceFactory.java:218)
    at com.hazelcast.instance.HazelcastInstanceFactory.newHazelcastInstance(HazelcastInstanceFactory.java:176)
    at com.hazelcast.test.TestHazelcastInstanceFactory.newHazelcastInstance(TestHazelcastInstanceFactory.java:116)
    at com.hazelcast.test.TestHazelcastInstanceFactory.newInstances(TestHazelcastInstanceFactory.java:166)
    at com.hazelcast.test.TestHazelcastInstanceFactory.newInstances(TestHazelcastInstanceFactory.java:172)
    at com.hazelcast.map.impl.mapstore.MapLoaderTest.testSenderAndBackupTerminates_AfterInitialLoad(MapLoaderTest.java:67)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-IbmJDK1.8/com.hazelcast$hazelcast/279/testReport/junit/com.hazelcast.map.impl.mapstore/MapLoaderTest/testSenderAndBackupTerminates_AfterInitialLoad/

### PR DESCRIPTION
**Pull request by tombujok**
Fixes #8508 spuriously failing

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/map/impl/mapstore/MapLoaderTest.java [hunk 1]
```diff
@@ -72,12 +74,26 @@ public void testSenderAndBackupTerminates_AfterInitialLoad() throws InterruptedE
...
         ownerAndReplicas[0].getLifecycleService().terminate();
         ownerAndReplicas[1].getLifecycleService().terminate();
+        waitUntilActive(ownerAndReplicas[3], 15);
 
         map = ownerAndReplicas[3].getMap(name);
         map.loadAll(false);
         assertEquals(DummyMapLoader.SIZE, map.size());
     }
 
+    private void waitUntilActive(HazelcastInstance hz, int timeoutSeconds) {
```
Comment (Donnerbart) [inline hazelcast/src/test/java/com/hazelcast/map/impl/mapstore/MapLoaderTest.java]: Should we put this in `HazelcastTestSupport`? It looks useful for other tests as well.

File: hazelcast/src/test/java/com/hazelcast/map/impl/mapstore/MapLoaderTest.java [hunk 2]
```diff
@@ -72,12 +74,26 @@ public void testSenderAndBackupTerminates_AfterInitialLoad() throws InterruptedE
...
+        int waited = 0;
+        while (!hz.getCluster().getClusterState().equals(ClusterState.ACTIVE)) {
+            try {
+                if (waited++ == timeoutSeconds) {
+                    break;
+                }
+                TimeUnit.SECONDS.sleep(1);
```
Comment (Donnerbart) [inline hazelcast/src/test/java/com/hazelcast/map/impl/mapstore/MapLoaderTest.java]: We have a method `sleepSeconds()` in `HazelcastTestSupport` which handles the `InterruptedException`.

File: hazelcast/src/test/java/com/hazelcast/map/impl/mapstore/MapLoaderTest.java (hunk 1 above)
Comment (tombujok) [inline hazelcast/src/test/java/com/hazelcast/map/impl/mapstore/MapLoaderTest.java]: 👍

File: hazelcast/src/test/java/com/hazelcast/map/impl/mapstore/MapLoaderTest.java (hunk 2 above)
Comment (tombujok) [inline hazelcast/src/test/java/com/hazelcast/map/impl/mapstore/MapLoaderTest.java]: 👍

File: hazelcast/src/test/java/com/hazelcast/map/impl/mapstore/MapLoaderTest.java [hunk 3]
```diff
@@ -72,6 +72,7 @@ public void testSenderAndBackupTerminates_AfterInitialLoad() throws InterruptedE
         HazelcastInstance[] ownerAndReplicas = findOwnerAndReplicas(instances, name);
         ownerAndReplicas[0].getLifecycleService().terminate();
         ownerAndReplicas[1].getLifecycleService().terminate();
+        waitUntilActive(ownerAndReplicas[3], 15);
```
Comment (mdogan) [inline hazelcast/src/test/java/com/hazelcast/map/impl/mapstore/MapLoaderTest.java]: I don't understand how this helps? When cluster gets out of `ACTIVE` state in this test?

Comment (mdogan) [inline hazelcast/src/test/java/com/hazelcast/map/impl/mapstore/MapLoaderTest.java]: yes, awaiting cluster size make sense. to ensure terminations of others are observed by alive one. 
there's already an assertion for that: `assertClusterSizeEventually()`

**Pull request merged**